### ISSUE TITLE
Dies after job received

### ISSUE THREAD

**Issue by johnwc** [doktor83/SRBMiner-Multi#52]
With this specific algo it just stops shortly after job received. I tried adding --extended-log, but nothing more gets logged.

```
Detecting OpenCL devices...

GPU0 : radeon_rx_580_series                  [ellesmere    ][MEM:  8192 MB][CU: 36][BUS: 03]
GPU1 : radeon_rx_580_series                  [ellesmere    ][MEM:  8192 MB][CU: 36][BUS: 04]
GPU2 : radeon_rx_580_series                  [ellesmere    ][MEM:  8192 MB][CU: 36][BUS: 06]
GPU3 : radeon_rx_570_series                  [ellesmere    ][MEM:  8192 MB][CU: 32][BUS: 07]

======================================================================
SRBMiner-MULTI 0.7.9

Press 's' to display stats
Press 'h' to display hashrate
Press 'p' to switch to the next pool
Press 'o' to switch to the previous pool
Press 0-9 to disable/enable GPU 0-9, shift+0-9 for GPU 10-19
Press + or - to change gpu boost value
======================================================================

Algorithm/s         : argon2d_dynamic [0.85% fee]
Gpu mining          : enabled
Cpu mining          : disabled
Gpu tweaking        : enabled
Gpu watchdog        : enabled

[2021-08-11 23:41:16] A0|GPU0 | [BUS:03] [I:      13728]
[2021-08-11 23:41:16] A0|GPU0 | [BUS:03] [I:      11936]
[2021-08-11 23:41:16] A0|GPU1 | [BUS:04] [I:      13728]
[2021-08-11 23:41:16] A0|GPU1 | [BUS:04] [I:      11936]
[2021-08-11 23:41:16] A0|GPU2 | [BUS:06] [I:      13728]
[2021-08-11 23:41:16] A0|GPU2 | [BUS:06] [I:      11936]
[2021-08-11 23:41:16] A0|GPU3 | [BUS:07] [I:      13728]
[2021-08-11 23:41:16] A0|GPU3 | [BUS:07] [I:      11936]
[2021-08-11 23:41:16] Loaded GPU tweaking library
[2021-08-11 23:41:17] Connected to argon2d500.na.mine.zpool.ca:4239 [argon2d_dynamic]
[2021-08-11 23:41:17] Difficulty 2.000000 [argon2d_dynamic][0]
[2021-08-11 23:41:17] Job received [493] block height 765698 [argon2d_dynamic][0]
[2021-08-11 23:41:18] Difficulty 2.000000 [argon2d_dynamic][0]
[2021-08-11 23:41:18] Job received [493] block height 765698 [argon2d_dynamic][0]

C:\SRBMiner-Multi-0-7-9>
```

**Comment by johnwc** [comment]
I have figured out that if I tell it to run with GPU 0 - 2, it runs fine. And if I tell it to run with just GPU 3, it runs fine. But, when it runs with a mix of 0-2 with 3, it dies.

**Comment by doktor83** [comment]
So running 2 instances of the miner should fix your issue : 
1 : gpu-id 0!1!2
2 : gpu-id 3

**Comment by johnwc** [comment]
That wasn't my solution, it was me trouble shooting what was causing it to fail on that algorithm. It runs on all four just fine with other algorithms.

I can't run two instances as normal everyday use, it's being managed by awesome miner.

This is the only miner we've run into this issue with. You can't look into this issue and try to resolve it in the miner?

**Comment by doktor83** [comment]
I can't reproduce your issue, share your config.

![Alternate image text](https://i.ibb.co/MgP3B9n/Untitled.jpg)

**Comment by johnwc** [comment]
This is the command line

`SRBMiner-MULTI.exe --extended-log --algorithm argon2d_dynamic --pool stratum+tcp://argon2d500.na.mine.zpool.ca:4239 --wallet *******.**** --password c=BTC --api-enable --api-port 4028 --disable-cpu`

**Comment by doktor83** [comment]
All same GPU's. [480 8g]. But that shouldn't matter, it's more than 4. I tried your config and it also works.
First i thought it might be that zpool sends some json that is misinterpreted or something but that's not the case.

**Comment by johnwc** [comment]
Well, notice that I have 3x580 and 1x570. The 570 only has 32 compute units compared to the 580 with 36 compute units.

It seems that if I set `--gpu-intensity 0`, it works. I saw a warning when using another algo, that stated to set that if the algo causes random crashes. It mentioned that using `--gpu-intensity 0` would cause the miner to create a thread per GPU.

**Comment by doktor83** [comment]
Ok, so it's basically the issue was that you had your intensities set too high?

**Comment by johnwc** [comment]
I did not set intensity at all, as you can see from the command I posted originally. I used that option to force it to create a thread per GPU, as I was thinking that there was a problem in the code for that algo when two cards had different CUs. Forcing a thread per GPU would alleviate that issue, and it seemed to do the trick.

**Comment by doktor83** [comment]
You could prob. use 2 threads per GPU too, just need to find the right intensity value.

**Comment by johnwc** [comment]
Well, I assumed that when it said 1 thread per GPU it was creating 4 threads instead of 1. I thought i was causing it to go from a single thread that controlled all GPUs, to 4 threads that each controlled a single GPU. I thought i was forcing a thread not to share GPUs.

I now am not sure what is going on then when that 570 GPU is added to the 580s. Is there a way to tell it intensity level per GPU?

**Comment by doktor83** [comment]
It's what it was written , those are threads per GPU. So 1 worker thread per GPU. Means 4 total if you use 4 GPU's.
Please check --gpu-intensity parameter it does exactly what you want. Even better, check the **examples** that are in the examples folder that comes in the archive with the miner.

**Comment by doktor83** [comment]
It's what it was written , those are threads per GPU. So 1 worker thread per GPU. Means 4 total if you use 4 GPU's.
Please check --gpu-intensity parameter it does exactly what you want. Even better, check the **examples** that are in the examples folder that comes in the archive with the miner.

Also im going to close this cause actually there is no issue 'Dies after job received'. and i dont want this ticket to become a tutorial on how to set up SRBMiner.

**Comment by johnwc** [comment]
This issue is not closed nor is this even close to being a tutorial... There is still an issue related to this algo and using GPUs with different compute units.

What is the default intensity when it is not set on the command line?

**Comment by doktor83** [comment]
Sorry man but i don't have time for this. 
You are asking me basic things (how to set intensity for a gpu), even though there are examples, readme's there, so yes - this would become a tutorial on how to use miner. Also the default intensity values are printed when you start the miner, and look like :

_A0|GPU0 | [BUS:03] [I:      13728]_

And for the end, believe it or not but you can mix different GPU's with different number of CU's and it will work. 
You went in a totally wrong direction here with your assumptions. What prob. just happens is that miner gives you too high
default values for the intensity , or you don't have enough virtual mem set in windows. That's all.

**Comment by johnwc** [comment]
Ok, continue to believe that it is not an issue with your miner, when it works with every other miner without issue or tweaking. I am attempting to try to help you work through why your miner is crashing OoB.

It sounds like your miner has tweaks set possibly too high that may cause some algo to crash... without even a error message... Shouldn't the miner be configured with more conservative settings that are more to work OoB perfectly. Then if the end-user wants to tweak them to have it run better, they do that? So that if it crashes, it is on the end user's edits pushing the miner/GPU to it's limit.

I can assure you that I am not the only one with this issue. I am just one of the few that doesn't just stop using a miner and go back to one that actually works. I try to reach out to the developer and report issues. I have been a developer for 20+ years, so I know that reports like this is the only way software gets improved. The developers needs to hear from the community about issues outside their controlled environment. BUT... the developer has to also be receptible that their software is not perfect and there could be minor bugs in it that need to be at least looked at. You seem to refuse to look into the code for this algo all together and just blow off that it has to be end user issue...

Also, a bit of background. We have been mining with 400+ GPU cards & 500 ASIC for over 8 years. We have worked with almost every miner/GPU that has come and gone. We know how to setup miners for correct mining in Windows/Linux. So... we have the correct amount of virt mem set.

**Comment by doktor83** [comment]
Oh  i fixed a lot of bugs and issues with the help of my users - i listen to people, but you are the kind of guy that's hard communicating with because you sound like you know everything better, don't accept what i tell you even when i proved you are wrong. Usually guys that are '100 years in this and have factories of GPU's' are like you.. :) No offense.

I just don't like when people behave like this, and i don't like to repeat the same thing over and over, when you are not listening.

You said _'There is still an issue related to this algo and using GPUs with different compute units'_ . I proved you are wrong, i even attached you a screenshot from my mixed GPU's rig.

We figured out that the auto intensity value you get isn't too good for your system. That can happen, there are a lot of GPU's, systems, setups out there and i just can't test every possible combination.

Because of the too high intensity value/s, probably too much memory is tried to be allocated. This causes a crash somewhere, maybe in the video driver, maybe in some other system call. Software crashes. Probably i could catch this somewhere but im not. Maybe i can't catch the exception cause its from a system component.

Anyways, i have a discord channel where i, myself, do some support. It's limited cause idon't have minions like some other miner devs do, and i have other day job beside miner development.
